### ISSUE TITLE
Allow Terra to initialize when LLVM is not build with NVPTX backend

### ISSUE THREAD

**Issue by elliottslaughter** [terralang/terra#446]
Right now we have a hard dependency on the NVPTX backend of LLVM even when CUDA is not enabled in Terra. This may potentially limit what LLVM binaries we can compile against. If we make the initialization conditional on CUDA support being enabled then we could build with more pre-built LLVM binaries.

Failure looks like:

```
terra/src/terralib.lua:747: failed to initialize target for LLVM Triple: nvptx64-nvidia-cuda (No available targets are compatible with this triple.)
```

**Comment by cata0309** [comment]
Can this dependency be removed before version is released? A lot of people will consider using it and will have this problem when trying terra.

**Comment by elliottslaughter** [comment]
As an update, we only initialize the target when `terralib.cudahome` is set, but that is currently unconditional on Linux. We should really make it smarter, because it's not as if we'll use this target if Terra isn't even built with CUDA.

I've added this to #492.

**Comment by elliottslaughter** [comment]
This was fixed in #541.
